src/lint/lint.nim

```diff
@@ -28,16 +44,11 @@ proc conceptExerciseFilesExist(trackDir: string): bool =
     ]
 
   let conceptDir = trackDir / "exercises" / "concept"
-  result = true
-
-  if dirExists(conceptDir):
-    for dir in getSortedSubDirs(conceptDir):
-      for conceptExerciseFile in conceptExerciseFiles:
-        let path = dir / conceptExerciseFile
-        if not fileExists(path):
-          writeError("Missing file", path)
+  result = subdirsContain(conceptDir, conceptExerciseFiles)
```

Comment:
A general question: when should one assign to `result` instead of doing `return`?

Reply:
That's a good question. It's partly a matter of personal taste.

The [official style guide](https://nim-lang.github.io/Nim/nep1.html#introduction-coding-conventions) contains:
> The `return` statement should ideally be used when its control-flow properties are required. Use a procedure's implicit `result` variable whenever possible. This improves readability.

That's a common style, which optimizes for making immediate `return` obvious. However, Nim style is still evolving: some people argue that we should use `return` more often, and use `result =` only when building up a return value over multiple lines.

We also have the option of omitting `result =` or `return` entirely here, writing just
```Nim
subdirsContain(conceptDir, conceptExerciseFiles)
```

The parent proc then returns the result of that call (the compiler enforces that it returns a `bool` as declared in the proc signature).

That style can improve readability/robustness in some places, but here I find it clearest to assign to `result` - we're doing the call because we're interested in its return value, rather than to perform some other action.

Let me know if I can clarify further.

You might like to read https://peterme.net/tips-and-tricks-with-implicit-return-in-nim.html - probably one of the best discussions on this topic.

Reply:
That _does_ help! Thanks